pkgs/by-name/si/siyuan/package.nix

```
@@ -117,12 +132,26 @@ stdenv.mkDerivation (finalAttrs: {
         --chdir $out/share/siyuan/resources \
         --add-flags $out/share/siyuan/resources/app \
         --set ELECTRON_FORCE_IS_PACKAGED 1 \
-        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations --enable-wayland-ime}}" \
         --inherit-argv0
 
     runHook postInstall
   '';
 
+  icon = fetchurl {
+    url = "https://raw.githubusercontent.com/siyuan-note/siyuan/master/app/src/assets/icon.svg";
+    hash = "sha256-S4YCU3wi6sgdKJyfsSL1T6dzZRfasVf92FD2uoHCwWo=";
+  };
+
+  postInstall = ''
+    for i in 16 24 48 64 96 128 256 512; do
+      mkdir -p $out/share/icons/hicolor/''${i}x''${i}/apps
+      convert -background none -resize ''${i}x''${i} ${icon} $out/share/icons/hicolor/''${i}x''${i}/apps/${pname}.png
+    done
+  '';
```

```suggestion
  postInstall = ''
    install -Dm644 src/assets/icon.svg $out/share/icons/hicolor/scalable/apps/siyuan.svg
  '';
```

This is probably much easier, no need to generate pngs